instrumentation/kafka/kafka-clients/kafka-clients-common/library/src/main/java/io/opentelemetry/instrumentation/kafkaclients/OpenTelemetryKafkaMetrics.java

```diff
@@ -0,0 +1,230 @@
+ * //    config.put(ProducerConfig.METRIC_REPORTER_CLASSES_CONFIG, OpenTelemetryKafkaMetrics.class.getName());
+ * //    config.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, ...);
+ * //    ...
+ * //    try (KafkaProducer<byte[], byte[]> producer = new KafkaProducer<>(config)) { ... }
+ * }</pre>
+ */
+public class OpenTelemetryKafkaMetrics implements MetricsReporter {
+
+  private static final Logger logger = Logger.getLogger(OpenTelemetryKafkaMetrics.class.getName());
+  private static final Set<KafkaMetricId> seenMetrics = ConcurrentHashMap.newKeySet();
+
+  @Nullable private static volatile Meter meter;
+
+  private static final Map<RegisteredInstrument, AutoCloseable> instrumentMap =
+      new ConcurrentHashMap<>();
+
+  /**
+   * Setup OpenTelemetry. This should be called as early in the application lifecycle as possible.
+   * Kafka metrics that are observed before this is called may not be bridged.
+   */
+  public static void setOpenTelemetry(OpenTelemetry openTelemetry) {
+    meter = openTelemetry.getMeter("io.opentelemetry.kafka-clients");
+  }
+
+  /**
+   * Reset for test by reseting the {@link #meter} to {@code null} and closing all registered
+   * instruments.
+   */
+  static void resetForTest() {
+    meter = null;
+    closeAllInstruments();
+  }
+
+  /**
+   * Print a table mapping kafka metrics to equivalent OpenTelemetry metrics, in markdown format.
+   */
+  static void printMappingTable() {
+    StringBuilder sb = new StringBuilder();
+    // Append table headers
+    sb.append(
+            "| # | Metric Group | Metric Name | Attribute Keys | Instrument Name | Instrument Description | Instrument Type |")
+        .append(lineSeparator())
+        .append(
+            "|---|--------------|-------------|----------------|-----------------|------------------------|-----------------|")
+        .append(lineSeparator());
+    Map<String, List<KafkaMetricId>> kafkaMetricsByGroup =
+        seenMetrics.stream().collect(groupingBy(KafkaMetricId::getGroup));
+    int count = 1;
+    // Iterate through groups in alpha order
+    for (String group : kafkaMetricsByGroup.keySet().stream().sorted().collect(toList())) {
+      List<KafkaMetricId> kafkaMetricIds =
+          kafkaMetricsByGroup.get(group).stream()
+              .sorted(comparing(KafkaMetricId::getName))
+              .collect(toList());
+      // Iterate through metrics in alpha order by name
+      for (KafkaMetricId kafkaMetricId : kafkaMetricIds) {
+        Optional<InstrumentDescriptor> descriptor =
+            Optional.ofNullable(KafkaMetricRegistry.getInstrumentDescriptor(kafkaMetricId));
+        // Append table row
+        sb.append(
+            String.format(
+                "| %s | %s | %s | %s | %s | %s | %s |%n",
+                count,
+                "`" + group + "`",
+                "`" + kafkaMetricId.getName() + "`",
+                String.join(",", kafkaMetricId.getAttributeKeys()),
```

Comment:
Nit: attribute names could be quoted too
```suggestion
                kafkaMetricId.getAttributeKeys()
                    .stream()
                    .map(key -> "`" + key + "`")
                    .collect(Collectors.joining(", ")),
```